cosipy/polarization/polarization_angle.py

```diff
@@ -100,4 +101,77 @@ def transform_to(self, convention, *args, **kwargs):
                                  self.skycoord,
                                  convention = convention2)
 
+    @classmethod
+    def from_scattering_direction(cls, psichi, source_coord, convention):
+        """
+        Calculate the azimuthal scattering angle of a scattered photon.
+        
+        Parameters
+        ----------
+        psichi : astropy.coordinates.SkyCoord
+            Scattered photon direction
+        source_coord : astropy.coordinates.SkyCoord
+            Source direction
+        convention :
+            cosipy.polarization.PolarizationConvention
+
+        Returns
+        -------
+        azimuthal_scattering_angle : cosipy.polarization.PolarizationAngle
+            Azimuthal scattering angle
+        """
+
+        if convention.frame.name == 'spacecraftframe':
+
+            source_coord = source_coord.transform_to(SpacecraftFrame(attitude=source_coord.attitude))
+            psichi = psichi.transform_to(SpacecraftFrame(attitude=source_coord.attitude))
+
+        elif convention.frame.name == 'icrs':
+
+            source_coord = source_coord.transform_to('icrs')
+            psichi = psichi.transform_to('icrs')
+
+        elif convention.frame.name == 'galactic':
+
+            source_coord = source_coord.transform_to('galactic')
+            psichi = psichi.transform_to('galactic')
+
+        else:
+
+            raise RuntimeError('Unsupported polarization convention frame "' + convention.frame.name + '"')
```

Comment:
I think you can change this by 

```
source_coord = source_coord.transform_to(convention.frame)
psichi = psichi.transform_to(convention.frame)
```
And let SkyCoord handle the transformation.